docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,517 @@
+# NodeAgentConfig configuration with restic/kopia
+Date: 2023-07-25
+
+## Abstract
+
+New configuration for the file system backup & restore that will be used by the OADP and allow to choose Restic or Kopia uploader type.
+
+## Background
+
+For the file system backup and restore the Velero may use Restic or Kopia as the uploader mechanism. A number of tests were [performed](https://velero.io/docs/main/performance-guidance/) by the Velero community to compare those mechanisms. In many cases Kopia uploader is a much better performing mechanism and as such should be added to the OADP operator.
+
+Please refer to the upstream [kopia uploader integration design](https://github.com/vmware-tanzu/velero/blob/main/design/unified-repo-and-kopia-integration/unified-repo-and-kopia-integration.md) for underlying Velero design and backup & restore workflow.
+
+Upstream [kopia](https://github.com/kopia/kopia) project which is used by Velero and configured by the OADP as this design proposes.
+
+## Goals
+
+- A new option `nodeagentconfig` to allow configuration of restic or kopia uploader
+- Backwards compatibility with the current OADP configuration schema options, namely `restic`
+- Preparation for the future deprecation of the current `restic` configuration option (from OADP 1.4+)
+- Allow new schema(s) to be used by the datamover node agent
+
+## Non Goals
+- Enablment of datamover node agent
```

Comment:
Ok will remove that. My understanding was different. Non goals is that even if they *will* allow enablment, the goal of this proposal is to focus on kopia and do not use it for datamover. If it's a side-effect or design to fit the future purpose it's ok to have it, but it's not part of kopia enablment design.

Reply:
@mpryc Since datamover requires node agent configuration, I think we should explicitly consider datamover enablement in scope. That's part of why we renamed it "nodeAgentConfig" from "fsBackupConfig".

Reply:
perhaps doesn't have to be done and tested in part-1 implementation?

Reply:
ok moved to the goals